packages/open-truss/src/configuration/engine-v1/RenderConfig.tsx

```diff
@@ -27,14 +27,14 @@ export function RenderConfig({
 }): React.JSX.Element {
   _COMPONENTS = COMPONENTS
   // Runs validations in config-schemas
-  const result = WorkflowV1Shape.safeParse(config)
-  if (!result.success) {
-    // TODO for now just raise any config validation errors.
-    // We likely want to render something nicer eventually.
-    throw result.error
-  }
+  // const result = WorkflowV1Shape.safeParse(config)
+  // if (!result.success) {
+  //   // TODO for now just raise any config validation errors.
+  //   // We likely want to render something nicer eventually.
+  //   throw result.error
+  // }
```

Comment:
Like I said in https://github.com/open-truss/open-truss/issues/92#issuecomment-1922844648, I think it would be best to move most config validation to the frame level so we can do things like I've done this in PR. Would love to hear what you all think!

Reply:
Oh interesting. So you're thinking at every `Frame` we call `FrameV1Shape.safeParse(config)` on the current node of the config and render the errors accordingly?

Reply:
I think so, yea! There might be a top level validation too, but in my opinion it shouldn't validate any frames.

Reply:
I think a pro / con might be
- pro 
  - will have `configPath` and can show exact config that failed
  - can render an error right on the frame in question
- con
  - we no longer will be front-loading validation. Maybe we can still do this for production and smoke tests, but do this PR's approach for config builder like tools

Reply:
Interesting con there! I think this PR could mostly stay the same but I'll:

1. Make the top-level validation always happen unless you tell it not to with a prop
2. Clean up how errors are thrown a bit so `configPath` doesn't need to be set at the error. This will let this system catch any errors that occur, not just configuration ones.